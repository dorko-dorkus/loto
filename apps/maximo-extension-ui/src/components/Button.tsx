import React, { ButtonHTMLAttributes } from 'react';
import clsx from 'clsx';

type ButtonProps = ButtonHTMLAttributes<HTMLButtonElement> & {
  className?: string;
};

export default function Button({ className, ...props }: ButtonProps) {
  return (
    <button
      className={clsx(
        'rounded-[var(--mxc-radius-sm)] bg-[var(--mxc-topbar-bg)] px-4 py-2 text-[var(--mxc-topbar-fg)] shadow-[var(--mxc-shadow-sm)]',
        className
      )}
      {...props}
    />
  );
}
