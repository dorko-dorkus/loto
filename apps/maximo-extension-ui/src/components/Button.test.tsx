import { render } from '@testing-library/react';
import { test, expect } from 'vitest';
import Button from './Button';

test('matches snapshot', () => {
  const { container } = render(<Button>Click</Button>);
  expect(container.firstChild).toMatchSnapshot();
});
