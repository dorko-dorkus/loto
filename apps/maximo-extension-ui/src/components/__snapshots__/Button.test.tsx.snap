// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<button
  class="rounded-[var(--mxc-radius-sm)] bg-[var(--mxc-topbar-bg)] px-4 py-2 text-[var(--mxc-topbar-fg)] shadow-[var(--mxc-shadow-sm)]"
>
  Click
</button>
`;
